Sources/SwiftKafka/KafkaConsumerMessage.swift

```diff
@@ -0,0 +1,78 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import struct Foundation.Data
+
+/// Messages that are received from the Kafka cluster.
+public struct KafkaConsumerMessage {
+    let topic: String
+    let partition: Int32
+    let key: Data?
+    let value: Data
+    let offset: Int64
+
+    // TODO: copy on write, does this even make sense as we read-only?
```

Comment:
We agreed on also having a copy-on-write mechanism for `KafkaConsumerMessage`. As Kafka messages are just read, why shouldn't we keep it as a read-only `struct` or have a `struct` that references an `_Internal` class that contains all the data?